### ISSUE TITLE
Question: Weights & detected descriptors

### ISSUE THREAD

**Issue by MrParano1d** [justadudewhohacks/face-api.js#214]
I'm pretty new to the face-recognition & neural network universe. I started with face-recognition.js for my first test project because the api looked simple to me. It also worked very well for me until I noticed the obsolete-message.

I've two questions:

1) What are weight models? Is it an manifest which gives the neural nets the necessary information for face recognition / landmarks / etc or does it already contain some information of persons? Like the one of the bbt people. If it contains people information, how do I train my own weights? (Which probably would answer my second question)
2) I've a database with images of people. I don't want to detect the faces every run from skratch. How do I save the results? In face-recognition.js I just could write `recognizer.serialize()` and save it to a file / database.
How can I achieve this with face-api.js?

Sorry for the dumb questions - as a beginner in this topic I'm totally confused right now.

None the less - thanks for this awesome library!

EDIT:
to 2) The detection and recognization is already working. It's just that I don't know how I can cache / save the parsed descriptors for a later use. I just wanna compare my known images with the uploaded photo.

**Comment by justadudewhohacks** [comment]
Hi,

> 1. What are weight models? Is it an manifest which gives the neural nets the necessary information for face recognition / landmarks / etc or does it already contain some information of persons? Like the one of the bbt people. If it contains people information, how do I train my own weights? (Which probably would answer my second question)

Think of a neural network as a very complex function with a lot of parameters. Model weights correspond to these parameters and since there are usually are a ton of them, we store them in a binary format and the value of every parameter is figured out during the learning process. They are learned by giving the neural net an input, forwarding it through the network and afterwards telling the network how wrong the predicted output is from what the real output should be (loss). Based on the loss value, all model weights in the network are updated in a process called backpropagation.

> 2\. I've a database with images of people. I don't want to detect the faces every run from skratch. How do I save the results? In face-recognition.js I just could write `recognizer.serialize()` and save it to a file / database.
>     How can I achieve this with face-api.js?

All that you have done in face-recognition.js with `recognizer.serialize()` saving the face descriptors. You would simply store the descriptors of each person in a database along with the persons name or something. The descriptors are just arrays with 128 entries.

**Comment by MrParano1d** [comment]
Thanks for your answer! That helped me alot.

I was thinking about to save the Float32Array from the descriptors but I wasn't sure if it's the correct way to go.
